jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java

```diff
@@ -64,10 +70,25 @@ public void init() throws Exception
     {
         _server = new Server();
 
+        _server.addBean(new ByteBufferPool()
+        {
+            @Override
+            public ByteBuffer acquire(int size, boolean direct)
+            {
+                return direct ? BufferUtil.allocateDirect(size) : BufferUtil.allocate(size);
+            }
+
+            @Override
+            public void release(ByteBuffer buffer)
+            {
+            }
+        });
```

Comment:
With this, this test is testing conditions that will never be the ones with the normal Jetty code, rendering this test useless.

Reply:
Because it is testing aggregation, which is dependent on the actual size of the buffer.  As our buffer pools can give back buffers with larger capacities, this test uses a simple buffer pool that gives back a buffer of exact size and thus allows the tests to be determinant about if they do or do not aggregate/commit etc.

Reply:
The real buffer pool behaviors are one of the reasons the original code failed in a production capacity.  I think @sbordet has a point, this should use a real world scenario, not a pie-in-the-sky wishful scenario.

Reply:
This is not pie in the sky wishful thinking.  It is a deterministic test case that actually tests what it thinks it is testing, not some random behaviour that is dependent on a buffer pool beyond the control of the test.

Reply:
OK, then add an alternate test that tests the opposite case.

A new ByteBufferPool implementation that always returns 4x the size requested.

Reply:
But that is not what I'm testing.  The tests only need to know if a write is bigger, smaller or the same as the buffers. It is relative size not absolute size that is important here.   If I had a 4x buffer returning pool, then I'd just have to alter the test case to write 4x as much data in each write to test the scenarios that are needed to test.

The point being that the HttpOutputTest is providing very good (well above average) coverage of HttpOutput.   The purpose here is not to test the buffer pools